WEBAPI.Infrastructure/DB/ApplicationDbContext.cs

```diff
@@ -0,0 +1,236 @@
+
+                    command.Parameters.Add(new SqlParameter("@Email", SqlDbType.NVarChar) { Value = user.Email });
+                    command.Parameters.Add(new SqlParameter("@Username", SqlDbType.NVarChar) { Value = user.Username });
+                    command.Parameters.Add(new SqlParameter("@PasswordHash", SqlDbType.NVarChar) { Value = hashedPassword });
+
+                    int rowsAffected = command.ExecuteNonQuery();
+                    if (rowsAffected > 0)
+                    {
+                        return "User registered successfully";
+                    }
+                    else
+                    {
+                        return "User registration failed";
+                    }
+                }
+            }
+        }
+        private static List<T> ConvertDataTable<T>(DataTable dt)
+        {
+            List<T> data = new List<T>();
+            foreach (DataRow row in dt.Rows)
+            {
+                T item = GetItem<T>(row);
+                data.Add(item);
+            }
+            return data;
+        }
+        private static T GetItem<T>(DataRow dr)
+        {
+            Type temp = typeof(T);
+            T obj = Activator.CreateInstance<T>();
+
+            foreach (DataColumn column in dr.Table.Columns)
+            {
+                foreach (PropertyInfo pro in temp.GetProperties())
+                {
+                    if (pro.Name.ToLower() == column.ColumnName.ToLower())
+                        pro.SetValue(obj, dr[column.ColumnName], null);
+                    else
+                        continue;
+                }
+            }
+            return obj;
+        }
+
+
+        //User Authentication and Authorization//
+
+
+        //public static string sqlDataSource = "Server=.;Database=WEBAPIAJAXPROJ;TrustServerCertificate=True;Integrated Security=true; MultipleActiveResultSets=true";
+
+        public DataTable GetData(string str, params IDataParameter[] parameters)
+        {
+            DataTable objresutl = new DataTable();
+
+            try
+            {
+                using (SqlConnection myCon = new SqlConnection(ConnectionString))
+                {
+                    myCon.Open();
+                    using (SqlCommand myCommand = new SqlCommand(str, myCon))
+                    {
+                        if (parameters != null)
+                        {
+                            myCommand.Parameters.AddRange(parameters);
+                        }
+
+                        using (SqlDataReader myReader = myCommand.ExecuteReader())
+                        {
+                            objresutl.Load(myReader);
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                // Handle exceptions appropriately
+            }
+
+            return objresutl;
+        }
+
+        public List<T> ConvertDataTableToList<T>(DataTable dt) where T : new()
```

Comment:
Are not these duplicate ConvertDataTable<T>() and ConvertDataTableToList<T>

Reply:
Yes they were duplicate, removed One.